Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -217,12 +217,24 @@ public async Task Push(Image x, string name, string? tag, string baseName, Actio
         manifestUploadContent.Headers.ContentType = new MediaTypeHeaderValue(DockerManifestV2);
         var putResponse = await client.PutAsync(new Uri(BaseUri, $"/v2/{name}/manifests/{x.GetDigest(x.manifest)}"), manifestUploadContent);
         string putresponsestr = await putResponse.Content.ReadAsStringAsync();
-        putResponse.EnsureSuccessStatusCode();
+
+        if (!putResponse.IsSuccessStatusCode)
```

Comment:
Can you pull this better error handling stuff into its own PR? I love it but would like to review separately if it's not too hard.